inbox/api/jsc_api.py

```diff
@@ -118,3 +120,32 @@ def auth_callback():
         })
 
         return make_response( (resp, 201, { 'Content-Type': 'application/json' }))
+
+@app.route('/retry_sync', methods=['POST'])
+def retry_sync():
+    g.parser.add_argument('namespace_public_id', required=True, type=valid_public_id, location='form')
+
+    args = strict_parse_args(g.parser, request.args)
+    shard = (args.get('target') or 0) >> 48
+
+    with session_scope(shard) as db_session:
+        try:
+            namespace = db_session.query(Namespace) \
+                .filter(Namespace.public_id == args['namespace_public_id']).first()
+
+            if namespace is None:
+                resp = simplejson.dumps({ 'message': 'Namespace do not exists', 'type': 'custom_api_error' })
+                return make_response((resp, 400, { 'Content-Type': 'application/json' }))
+
+            account = namespace.account
+            resp = simplejson.dumps({
```

Comment:
I think this could be moved next to the `make_response` call, just to better organize the code in logical parts.

Another thing, just out of curiosity: are you returning `sync_status['sync_error']` so we can check the retry worked?